mux_client_impl/src/error.rs

```diff
@@ -0,0 +1,180 @@
+use std::io;
+
+use openssh_mux_client::connection;
+use thiserror::Error;
+
+/// Errors that occur when interacting with a remote process.
+#[derive(Debug, Error)]
+pub enum Error {
```

Comment:
The `Error` type should definitely be shared among the impls.

Reply:
Yep, it’s definitely doable and I will merge them.

Reply:
A unified `Error` is implemented.